File: C++/MakananIkan.cpp
#include <bits/stdc++.h>
#include "MakananIkan.hpp"
#include "Aquarium.hpp"

const int MISpeed = 1;       // kecepatan jatuh makanan ikan
const int minMIPrice = 20;   // harga minimum makanan ikan
const int maxMIPrice = 50;   // harga maksimum makanan ikan
const double topScreen = 100;

// ctor default
MakananIkan::MakananIkan() : ObjekMati(MISpeed){
    price = rand() % (maxMIPrice - minMIPrice + 1) + minMIPrice;    // random harga makanan ikan dari minMIPrice - maxMIPrice
    jenis = "Makanan Ikan";
}

// ctor user-define
MakananIkan::MakananIkan(double absis): ObjekMati(MISpeed){
    price = rand() % (maxMIPrice - minMIPrice + 1) + minMIPrice;    // random harga makanan ikan dari minMIPrice - maxMIPrice
    jenis = "Makanan Ikan";
    Point posisi(absis,topScreen);                                  // posisi akan mulai dijatuhkan makanan
    setPosisi(posisi);
}

// getter price
double MakananIkan::getPrice(){
    return price;
}

// setter price
void MakananIkan::setPrice(double newPrice){
    price = newPrice;
}

File: Java/MakananIkan.java
import java.util.Random;

public class MakananIkan extends ObjekMati {

  private final int kecepatanMakananIkan = 1;
  private final int constHarga = 10;
  private int harga;

  /**
   * konstruktor default makanan ikan.
   */
  public MakananIkan() {
    super(1);
    harga = constHarga;
    jenis = "Makanan Ikan";
  }

  /**
   * konstruktor user-defined makanan ikan.
   * @param posisi posisi makananIkan saat diciptakan
   */
  public MakananIkan(Point posisi) {
    super(1);//kecepatanMakananIkan
    this.harga = constHarga;
    this.jenis = "Makanan Ikan";
    this.posisi = posisi;
  }

  /**
   * getter atribut harga makananIkan.
   * @return harga makananIkan int
   */
  public int getHarga() {
    return this.harga;
  }

  /**
   * setter atribut harga makananIkan
   * @param harga nilai harga baru
   */
  public void setHarga(int harga) {
    this.harga = harga;
  }
}
